src/fhirtypes/ValueSet.ts

```diff
@@ -0,0 +1,72 @@
+import { Meta } from './specialTypes';
+
+import { Extension } from '../fshtypes';
+import { Narrative, Resource, Identifier, CodeableConcept, Coding } from './dataTypes';
+import { ContactDetail, UsageContext } from './metaDataTypes';
+
+/**
+ * Class representing a FHIR R4 ValueSet.
+ *
+ * @see {@link http://hl7.org/fhir/R4/valueset.html | FHIR ValueSet}
+ */
+
+export class ValueSet {
+  id: string;
+  meta: Meta;
+  implicitRules: string;
+  language: string;
+  text: Narrative;
+  contained: Resource[];
+  extension: Extension[];
+  modifierExtension: Extension[];
+  url: string;
+  identifier: Identifier[];
+  version: string;
+  name: string;
+  title: string;
+  status: string;
+  experimental: boolean;
+  date: string;
+  publisher: string;
+  contact: ContactDetail[];
+  description: string;
+  useContext: UsageContext[];
+  jurisdiction: CodeableConcept[];
+  immutable: boolean;
+  purpose: string;
+  copyright: string;
+  compose: ValueSetLogicalDefinition;
+}
+
+export type ValueSetLogicalDefinition = {
+  lockedDate?: string;
```

Comment:
So I'm noticing that we're a little inconsistent in using `?` to indicate optionality.  We don't use it in any of the properties directly on `ValueSet`, but we are using it here.  I think the same can be said for how we did `StructureDefinition`.

It doesn't need to be in this PR, but I think we should consider adding the optionality flags to the higher-level properties too (here and in other fhir classes we've already written).